developer/src/kmc-kmn/src/compiler/compiler.ts

```diff
@@ -136,30 +131,91 @@ export class KmnCompiler {
     // TODO: use callbacks for file access -- so kmc-kmn is entirely fs agnostic
     let result = this.runCompiler(infile, outfile, options);
     delete (globalThis as any)[this.callbackName];
-    return result;
+    //TODO: write the file out!
+    if(result) {
+      if(result.kmx) {
+        this.callbacks.fs.writeFileSync(result.kmx.filename, result.kmx.data);
+      }
+      if(result.kvk) {
+        this.callbacks.fs.writeFileSync(result.kvk.filename, result.kvk.data);
+      }
+      if(result.js) {
+        this.callbacks.fs.writeFileSync(result.js.filename, result.js.data);
+      }
+    }
+    return !!result;
   }
 
   private compilerMessageCallback = (line: number, code: number, msg: string): number => {
     this.callbacks.reportMessage(mapErrorFromKmcmplib(line, code, msg));
     return 1;
   }
 
-  private runCompiler(infile: string, outfile: string, options: CompilerOptions): boolean {
+  private runCompiler(infile: string, outfile: string, options: CompilerOptions): CompilerResult {
     try {
-      if (!this.wasm.setCompilerOptions(options.shouldAddCompilerVersion ? 1 : 0)) {
-        this.callbacks.reportMessage(CompilerMessages.Fatal_UnableToSetCompilerOptions());
+      let result: CompilerResult = {};
+      let wasm_interface = new this.wasm.Module.CompilerInterface();
+      let wasm_result = null;
+      try {
```

Comment:
Yes, I see I can combine the inner-and-outer blocks to `try`/`catch`/`finally`. The outer one is a catch-all for exceptions because we want to handle unexpected exceptions from the WASM code. And ... I note that kmc is not sentryfied which is _not_ a good thing. Will open a new issue to track that.